client/components/AllPuzzles.js

```diff
@@ -0,0 +1,43 @@
+import React from "react"
+import {connect} from "react-redux"
+import {fetchAllPuzzles} from "../store/puzzles"
+
+
+export class AllPuzzles extends React.Component{
+  constructor(props){
+    super(props)
+  }
+  componentDidMount(){
+    this.props.fetchAllPuzzles()
+
+  }
+  render(){
+    allPuzzles= this.props.puzzle
+    return(
+      <div>
+        {
+          allPuzzles && allPuzzles.map(puzzle=>{
+            <div key={puzzle.id}>
+            <SinglePuzzle puzzle={puzzle}/>
```

Comment:
I think youll need to do a separate component for this. The "single puzzle view" that I made includes all the info... description, dimensions, etc. that won't necessarily be shown in the allPuzzles view

Reply:
Ill get that created now and add to the next pull request to display the name, picture, and cost, and I will rename it to avoid confusion